### PR TITLE
Add more details on documentation to the PR checklist

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit expands step 2 on updating documentation in the
CONTRIBUTING.md file's checklist for opening a PR. This was previously
missing an explanation about needing to update the `autosummary`
directives if a new class, function, or module is added otherwise it
won't be included in the documentation builds. In addition to fixing
this oversight a link is added to the contributing documentation's
section on the documentation structure to provide more details on how to
add/where to add new objects to the documentation build.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
Oh, I probably should add a note here that if the class, function, or module is user facing. We don't want to add things to the documentation if we don't intend it to be consumed externally.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +21,17 @@ please ensure that:
 2. The documentation has been updated accordingly. In particular, if a
    function or class has been modified during the PR, please update the
    *docstring* accordingly.
+
+   If your pull request is adding a new class, function, or module that is
+   intended to be user facing also ensure that you've also added those to a
```
Comment (1ucian0) [inline CONTRIBUTING.md:26]: Repetitive? 
```suggestion
   intended to be user facing ensure that you've also added those to a
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -21,6 +21,17 @@ please ensure that:
...
    *docstring* accordingly.
+
+   If your pull request is adding a new class, function, or module that is
+   intended to be user facing ensure that you've also added those to a
+   documentation `autosummary` index to include it in the api documentation.
+   For more details you can refer to:
+
+   https://qiskit.org/documentation/contributing_to_qiskit.html#documentation-structure
+
+   which documents how the documentation is structured and how you can add
```
Comment (javabster) [inline CONTRIBUTING.md:32]: I think the last line of this sentence may be a bit redundant imho. I think it's fine to just say 'for more details you can refer to <link>.'
This reduces the cognitive overload a bit, meaning people may be less likely to skip over.

Comment (javabster) [inline CONTRIBUTING.md:32]:
```suggestion
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,6 +21,17 @@ please ensure that:
...
+   new classes, functions, or modules to it.
```
Comment (javabster) [inline CONTRIBUTING.md:33]:
```suggestion
```

**Review by javabster — APPROVED** [review]
LGTM 🚀

**Review by levbishop — APPROVED** [review]

**Pull request merged**